casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -309,6 +309,7 @@ class RuntimeManagerImpl[F[_]: Concurrent] private[rholang] (
     def evalSingle(acc: Acc, deploy: DeployData): F[Acc] = {
       val (hash, deployResults) = acc
       for {
+        _                            <- runtime.space.reset(hash)
```

Comment:
I thought we call this outside of newEval. Why are we calling it again?

Reply:
Also I'm surprised the lack of this hasn't shown any major failures. Do you have an explanation for this?

Reply:
we need to call it between the (faulty) deploys, not just before the newEval. Notice `evalSingle` is being `foldM`-ed to make `newEval` now.

My explanaion for lack of failures in tests is 'we don't check that an erroneous deploy doesn't leave any garbage in the TS'